design/20190708.certificate-request-crd.md

```diff
@@ -206,6 +206,22 @@ ownership of the default pool of issuers in the cert-manager project.
 Until the mutating webhook is fully implemented, we will handle defaulting
 internally in the controller.
 
+### CertificateRequest Annotations
+In order for `CertificateRequest` controllers to resolve requests, extra
+information may be needed that is not present in the API Spec. To pass on this
+information, a set of one or more annotations should be defined, with reliable
+value pairs. These annotations should be considered optional. Any
+`CertificiateRequest` controller that rely on these to function should fallback
+gracefully or be marked as failed in the event a required annotation is missing.
+The currently defined annotations are:
+
+- `certmanager.k8s.io/private-key-secret-name`: The name of the secret, in the
+  same namespace as the `CertificateRequest`, that stores the private key which
+  was used to sign the x509 certificate signing request. This is required by the
+ `SelfSigning` issuer to sign its own certificate. Currently the `Certificate`
+  controller adds this annotation to all `CertificateRequest` resources it
+  creates with the defined `SecretName` in the Spec of the `Certificate`.
```

Comment:
Maybe add "If it is not set, the self-signing issuer marks the CertificateRequest as failed"